lib/providers/map_data.dart

```diff
@@ -24,11 +24,23 @@ class MapData extends ChangeNotifier {
   LatLng _start;
   LatLng _end;
   String _mode;
+  String _duration;
+  String _distance;
```

Comment:
recurring variables as seen in itinerary.dart

Reply:
Agreed. Both itinerary and map_data seem to overlap with their responsibility and I think the Itinerary class needs to be looked over as it has some class attributes yet it's main functions are static.